formasaurus/formtype_model.py

```diff
@@ -80,22 +80,143 @@ def _create_feature_union(features):
     return make_union(*[make_pipeline(fe, vec) for name, fe, vec in features])
 
 
+_PROB_CLF_CLASS = LogisticRegression
+
+
+def _is_prob_clf(clf):
+    return isinstance(clf, _PROB_CLF_CLASS)
+
+
+def _get_prob_clf():
+    return _PROB_CLF_CLASS(penalty="l2", C=5, fit_intercept=True)
+
+
+def _get_non_prob_clf():
+    return LinearSVC(C=0.5, random_state=0, fit_intercept=True)
+
+
 def get_model(prob=True):
     """
     Return a default model.
     """
     # XXX: fit_intercept is False for easier model debugging.
     # Intercept is included as a regular feature ("Bias").
 
-    if prob:
-        clf = LogisticRegression(penalty="l2", C=5, fit_intercept=True)
-    else:
-        clf = LinearSVC(C=0.5, random_state=0, fit_intercept=True)
-
+    clf = _get_prob_clf() if prob else _get_non_prob_clf()
     fe = _create_feature_union(FEATURES)
     return make_pipeline(fe, clf)
 
 
+_VECTORIZER_INIT_ATTRIBUTES = {
+    "CountVectorizer": ("ngram_range", "min_df", "binary"),
+    "DictVectorizer": tuple(),
+    "TfidfVectorizer": ("ngram_range", "min_df", "binary", "stop_words", "analyzer"),
+}
+_VECTORIZER_NON_INIT_ATTRIBUTES = {
+    "CountVectorizer": ("vocabulary_",),
+    "DictVectorizer": ("feature_names_", "vocabulary_"),
+    "TfidfVectorizer": ("idf_", "vocabulary_"),
+}
+
+
+def to_dict(model):
```

Comment:
should we name the argument `pipeline` instead? It seems it assumes sklearn's Pipeline